pytorch_lightning/trainer/training_loop.py

```diff
@@ -396,6 +394,12 @@ def _process_training_step_output_1_0(self, training_step_output, split_batch):
             hiddens = training_step_output.pop('hiddens', None)
             result['extra'] = training_step_output
 
+            if loss is None:
```

Comment:
don't we support returning None anymore? I thought this was to skip the backward and optimiser step?

Reply:
Hey @justusschock 

On my understanding of the code, here is what currently supported:

+ None (It will skip backward, optimizer step, etc...)

+ torch.Tensor (Considered as loss)

+ dict 

`current behaviour`:
Pop loss key from dict, create result object with result.minize = loss, and break when creating closure

`new behaviour`:
Pop loss key from dict, make sure loss key with MisConfigurationExpection instead of breaking on closure creation.

 It is a little bit patching legacy code...

Reply:
@williamFalcon, do you validate with that ?